test/runner/job_test.exs

```diff
@@ -45,4 +19,253 @@ defmodule ElixirBench.Runner.JobTest do
+
+      assert_raise MatchError, fn ->
+        Job.ensure_no_other_jobs!()
+      end
+
+      Process.exit(pid, :kill)
+    end
+  end
+
+  describe "get_benchmars_output_path/1" do
+    test "return path for writing benchmark results" do
+      env_backup = Application.get_env(:runner, :benchmars_output_path)
+      Application.put_env(:runner, :benchmars_output_path, "/tmp")
+
+      assert "/tmp/test_job" = Job.get_benchmars_output_path(job_fixture())
+
+      Application.put_env(:runner, :benchmars_output_path, env_backup)
+    end
+
+    test "raise if output path not defined" do
+      env_backup = Application.get_env(:runner, :benchmars_output_path)
+      Application.delete_env(:runner, :benchmars_output_path)
+
+      assert_raise ArgumentError, fn ->
+        Job.get_benchmars_output_path(job_fixture())
+      end
+
+      Application.put_env(:runner, :benchmars_output_path, env_backup)
+    end
+  end
+
+  describe "get_compose_config/1" do
+    test "return docker-compose setup for simple job" do
+      job = job_fixture()
+
+      compose_config =
+        Job.get_compose_config(job)
+        |> Antidote.decode!()
+
+      assert %{
+               "services" => %{
+                 "runner" => %{
+                   "image" => "elixirbench/runner:1.5.2-20.1.2",
+                   "command" => "mix run bench/bench_helper.exs",
+                   "network_mode" => "host",
+                   "environment" => %{
+                     "ELIXIRBENCH_REPO_SLUG" => "elixir-ecto/ecto",
+                     "ELIXIRBENCH_REPO_BRANCH" => "mm/benches",
+                     "ELIXIRBENCH_REPO_COMMIT" => "2a5a8efbc3afee3c6893f4cba33679e98142df3f"
+                   },
+                   "volumes" => [
+                     "/tmp/benchmarks/test_job:/var/bench:Z"
+                   ]
+                 }
+               },
+               "version" => "3"
+             } = compose_config
+    end
+
+    test "return docker-compose setup for job with deps and environment variables" do
+      job = job_fixture() |> with_deps |> with_env_variables
+      timeout = 200
+
+      compose_config =
+        Job.get_compose_config(job)
+        |> Antidote.decode!()
+
+      %{"services" => services} = compose_config
+
+      assert %{"job_test_job_mysql" => _mysql} = services
+      assert %{"job_test_job_postgres" => _postgres} = services
+
+      %{"runner" => %{"command" => command}} = services
+
+      assert command =~
+               "wait-for.sh localhost:3306 -t #{timeout} -- wait-for.sh localhost:5432 -t #{
+                 timeout
+               } -- mix run bench/bench_helper.exs"
+
+      assert %{"runner" => %{"depends_on" => ["job_test_job_mysql", "job_test_job_postgres"]}} =
+               services
+
+      assert %{"runner" => %{"environment" => %{"PG_URL" => "postgres:postgres@localhost"}}} =
+               services
+
+      assert %{"runner" => %{"environment" => %{"MYSQL_URL" => "root@localhost"}}} = services
+    end
+  end
```

Comment:
I was unsure if your changes in `build_services` were ok in retaining behaviour, but these tests help my confidence - thank you :+1:

Reply:
I can make it more explicit and put the entire config within a pattern match